File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task132_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.IOException;
import java.util.Scanner;
import java.util.logging.FileHandler;
import java.util.logging.Logger;
import java.util.logging.SimpleFormatter;

public class SecurityEventMonitor {
    private static final Logger LOGGER = Logger.getLogger(SecurityEventMonitor.class.getName());

    public static void setupLogger() throws IOException {
        FileHandler fileHandler = new FileHandler("security_events.log", true);
        SimpleFormatter formatter = new SimpleFormatter();
        fileHandler.setFormatter(formatter);
        LOGGER.addHandler(fileHandler);
    }

    public static void logEvent(String eventType, String description) {
        LOGGER.info("Event Type: " + eventType + ", Description: " + description);
    }

    public static void monitorEvents() {
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.print("Enter event type (or \'q\' to quit): ");
            String eventType = scanner.nextLine();
            if (eventType.equalsIgnoreCase("q")) {
                break;
            }
            System.out.print("Enter event description: ");
            String description = scanner.nextLine();
            logEvent(eventType, description);
        }
        scanner.close();
    }

    public static void main(String[] args) {
        try {
            setupLogger();
            System.out.println("Security Event Monitoring System");
            monitorEvents();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task132_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#define MAX_LENGTH 256

void logEvent(const char* eventType, const char* description) {
    FILE* logFile = fopen("security_events.log", "a");
    if (logFile != NULL) {
        time_t now;
        time(&now);
        char* time_str = ctime(&now);
        time_str[strlen(time_str) - 1] = '\\0';  // Remove newline
        fprintf(logFile, "%s - INFO - Event Type: %s, Description: %s\\n", 
                time_str, eventType, description);
        fclose(logFile);
    } else {
        fprintf(stderr, "Unable to open log file!\\n");
    }
}

int main() {
    char eventType[MAX_LENGTH];
    char description[MAX_LENGTH];

    printf("Security Event Monitoring System\\n");

    while (1) {
        printf("Enter event type (or 'q' to quit): ");
        fgets(eventType, sizeof(eventType), stdin);
        eventType[strcspn(eventType, "\\n")] = 0;  // Remove newline

        if (strcmp(eventType, "q") == 0 || strcmp(eventType, "Q") == 0) {
            break;
        }

        printf("Enter event description: ");
        fgets(description, sizeof(description), stdin);
        description[strcspn(description, "\\n")] = 0;  // Remove newline

        logEvent(eventType, description);
        printf("Event logged successfully.\\n");
    }

    printf("Monitoring ended. Check 'security_events.log' for recorded events.\\n");
    return 0;
}
